src/argilla/server/search_engine/base.py

```diff
@@ -72,44 +72,57 @@
 ]
 
 
-class SuggestionFilterScope(BaseModel):
-    question: str
-    property: str
+class SuggestionFilterScope:
+    def __init__(self, question: str, property: str):
+        self.question = question
+        self.property = property
 
 
-class ResponseFilterScope(BaseModel):
-    question: Optional[str]
-    property: Optional[str]
+class ResponseFilterScope:
+    def __init__(self, question: Optional[str] = None, property: Optional[str] = None, user: Optional[User] = None):
+        self.question = question
+        self.property = property
+        self.user = user
 
 
-class MetadataFilterScope(BaseModel):
-    metadata_property: str
+class MetadataFilterScope:
+    def __init__(self, metadata_property: str):
+        self.metadata_property = metadata_property
 
 
-FilterScope = Union[SuggestionFilterScope, ResponseFilterScope, MetadataFilterScope]
+class RecordFilterScope:
+    def __init__(self, property: str):
+        self.property = property
 
 
-class TermsFilter(BaseModel):
-    scope: FilterScope
-    values: List[str]
+FilterScope = Union[SuggestionFilterScope, ResponseFilterScope, MetadataFilterScope, RecordFilterScope]
+
+
+class TermsFilter:
+    def __init__(self, scope: FilterScope, values: List[str]):
```

Comment:
I know `TermsFilter` handlers schema has `values` defined as `List[str]` but I'm thinking that maybe for `RatingQuestions` this could be not true. I mean `RatingQuestions` have `values` as `List[int]` so maybe we need to change it.

Reply:
I discussed this point with @frascuchon and we will use `RangeFilter`for `RatingQuestions` so by now this change is not necessary.